ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallViewState.kt

```diff
@@ -1,9 +1,17 @@
 package com.revenuecat.purchases.ui.revenuecatui
 
-import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.paywalls.PaywallData
 
-sealed class PaywallViewState {
+internal sealed class PaywallViewState {
     object Loading : PaywallViewState()
     data class Error(val errorMessage: String) : PaywallViewState()
-    data class Loaded(val offering: Offering) : PaywallViewState()
+    data class Template2(
```

Comment:
For now, we only have `Template2`. But ideally the UI is just driven from this state

Reply:
There's only one template so it's fine for now, but this is going to be duplicated for every template.

I'd recommend creating a `PaywallTemplate` `enum` and changing this to just `Template` containing a `PaywallTemplate`. Then you can switch over that template enum to determine which view to display.

Does that make sense?

Reply:
Yeah, that makes sense. We can do that refactor in future PRs 👍